libgalois/src/analytics/local_clustering_coefficient/local_clustering_coefficient.cpp

```diff
@@ -0,0 +1,365 @@
+          (*node_triangle_count)[vv] += 1;
+        }
+      }
+    }
+  }
+
+  /*
+ * Simple counting loop, instead of binary searching.
+ * It assumes that edgelist of each node is sorted.
+ * This uses a PerThreadStorage implementation.
+ */
+  void OrderedCountAlgo(Graph* graph) {
+    katana::PerThreadStorage<std::vector<uint64_t>>
+        per_thread_node_triangle_count;
+    uint64_t num_nodes = graph->size();
+    uint32_t num_threads = katana::getActiveThreads();
+
+    katana::do_all(
+        katana::iterate((uint32_t)0, num_threads), [&](uint32_t tid) {
+          per_thread_node_triangle_count.getRemote(tid)->resize(num_nodes, 0);
+        });
+
+    katana::do_all(
+        katana::iterate(*graph),
+        [&](const Node& n) {
+          OrderedCountFunc(
+              graph, n, &(*per_thread_node_triangle_count.getLocal()));
+        },
+        katana::chunk_size<kChunkSize>(), katana::steal(),
+        katana::loopname("TriangleCount_OrderedCountAlgo"));
+
+    katana::do_all(
+        katana::iterate(*graph),
+        [&](const Node& n) {
+          node_triangle_count_[n] = 0;
+          for (uint32_t i = 0; i < num_threads; i++) {
+            node_triangle_count_[n] +=
+                (*(per_thread_node_triangle_count.getRemote(i)))[n];
+          }
+        },
+        katana::chunk_size<kChunkSize>(), katana::steal(),
+        katana::loopname("TriangleCount_Reduce"));
+  }
+
+  void ComputeLocalClusteringCoefficient(Graph* graph) {
+    katana::do_all(katana::iterate(*graph), [&](Node n) {
+      auto degree =
+          std::distance(graph->edges(n).begin(), graph->edges(n).end());
+      graph->template GetData<NodeClusteringCoefficient>(n) =
+          ((double)(2 * node_triangle_count_[n])) / (degree * (degree - 1));
+    });
+
+    return;
+  }
+
+  katana::Result<void> operator()(
+      katana::PropertyGraph* pg, const std::string& output_property_name) {
+    if (auto result = katana::analytics::ConstructNodeProperties<NodeData>(
+            pg, {output_property_name});
+        !result) {
+      return result.error();
+    }
+
+    auto graph_result = Graph::Make(pg, {output_property_name}, {});
+    if (!graph_result) {
+      return graph_result.error();
+    }
+
+    Graph graph = graph_result.value();
+
+    katana::StatTimer execTime(
+        "LocalClusteringCoefficient", "LocalClusteringCoefficient");
+    execTime.start();
+
+    node_triangle_count_.allocateBlocked(graph.size());
+
+    // Calculate the number of triangles
+    // on each node
+    OrderedCountAlgo(&graph);
+
+    // Compute the clustering coefficient of each
+    // node based on the triangles.
+    ComputeLocalClusteringCoefficient(&graph);
+
+    node_triangle_count_.destroy();
+    node_triangle_count_.deallocate();
+
+    execTime.stop();
+
+    return katana::ResultSuccess();
+  }
+};
+}  // namespace
+
+template <typename Algorithm>
+katana::Result<void>
+katana::analytics::LocalClusteringCoefficientWithWrap(
```

Comment:
Can you please move the graph construction here and then pass the TypedGraph  to the `algo(...)` instead of propertyGraph? Similar to connected-components.